collateral.md

```diff
@@ -26,6 +32,63 @@ module COLLATERAL
         </flips>
       </collateral>
 
+    syntax MCDContract ::= GemContract
+    syntax GemContract ::= "Gem" Address
+    syntax MCDStep ::= GemContract "." GemStep [klabel(gemStep)]
+ // ------------------------------------------------------------
+    rule contract(Gem ACCTGEM . _) => Gem ACCTGEM
+    rule [[ address(Gem ACCTGEM) => ACCTGEM ]] <gem-addr> ACCTGEM </gem-addr>
+
+    syntax GemAuthStep
+    syntax GemStep ::= GemAuthStep
+    syntax AuthStep ::= GemContract "." GemAuthStep [klabel(gemStep)]
+ // -----------------------------------------------------------------
+    rule <k> Gem _ . _ => exception ... </k> [owise]
+
+    syntax GemStep ::= "transferFrom" Address Address Int
+ // -----------------------------------------------------
+    rule <k> Gem ACCTGEM . transferFrom ACCTSRC ACCTDST VALUE => . ... </k>
+         <gem>
+           <gem-addr> ACCTGEM </gem-addr>
+           <gem-balances>...
+             ACCTSRC |-> ( BALANCE_SRC => BALANCE_SRC -Int VALUE )
+             ACCTDST |-> ( BALANCE_DST => BALANCE_DST +Int VALUE )
+           ...</gem-balances>
+         </gem>
+      requires BALANCE_SRC >=Int VALUE
+
+    syntax GemStep ::= "move" Address Address Int
+ // ---------------------------------------------
+    rule Gem ACCTGEM . (move ACCTSRC ACCTDST VALUE => transferFrom ACCTSRC ACCTDST VALUE)
+
+    syntax GemStep ::= "transfer" Address Int
+ // -----------------------------------------
+    rule <k> Gem ACCTGEM . (transfer ACCTDST VALUE => transferFrom MSGSENDER ACCTDST VALUE) ... </k>
+         <msg-sender> MSGSENDER </msg-sender>
+
+    syntax GemStep ::= "mint" Address Int
+ // -------------------------------------
+    rule <k> Gem ACCTGEM . mint ACCTDST VALUE => . ... </k>
+         <gem>
+           <gem-addr> ACCTGEM </gem-addr>
+           <gem-balances>...
+             ACCTDST |-> ( BALANCE_DST => BALANCE_DST +Int VALUE )
+           ...</gem-balances>
+         </gem>
+      requires ACCTDST =/=Int 0
```

Comment:
Where is this constraint coming in? I don't see any reason that `ACCTDST` can't be `0`.

Also, we should check that `VALUE >=Int 0`.

Reply:
It came from a `require(sender != address(0))` statement in the ERC20 solidity file I referenced, but now that you mention it I guess it doesn't matter in this model.

Similarly, the ERC20 contract was using `uint` for the value, so it didn't need to make sure it's positive. I'll add in the check.

[The ERC20.sol I referenced](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol)